### ISSUE TITLE
As a user, I want to run repo mirroring services, so that the feature is enabled and works.

### ISSUE THREAD

**Issue by thomasmckay** [quay/quay-operator#64]
With quay-v3.1.0 there is a new feature named "repo mirroring". This feature defaults to off and thus must be enabled via the config.yaml/config app. In addition, by default the worker (aka service) that does the mirroring is not started in the "registry" container mode. Instead, a new entrypoint "repomirror" is available to run only the services necessary for this feature. The intent is that there will be both a "registry" pod and the "repomirror" pod running. The "repomirror" pod requires access to the database (like "registry" does), as well as access to external registries to be mirrored. An external endpoint is not required.

An alternate starting mechanism for "repomirror" is available to run the worker inside the "registry" container. To do this, the environment variable ```QUAY_OVERRIDE_SERVICES``` is given a value of ```repomirrorrowker=true```. This mode is intended to be for POC but is fully functional and may, in fact, be the most common mode for users.

For the quay-operator, I'd suggest two phase approach: First, add the env var method to allow usage. Next, add effectively a duplicate setup as "registry" but change entry point to "repomirror".

Differences in "repomirror" pod:

- At this time, there is no readiness/liveness endpoints so default them to be omitted. Leave the functionality, though, since upcoming release will support that.
- An external route to this pod is unnecessary at this time. A distant release may support this, though, so if it makes sense to leave support in code but disabled, that would not be wasted effort.
- Scaling, sizing, and other pod aspects should be available but separate from the "registry" pods (ie. they will be adjusted separately).

**Comment by kurtismullins** [comment]
@thomasmckay 

>At this time, there is no readiness/liveness endpoints so default them to be omitted. Leave the functionality, though, since upcoming release will support that.

Did Quay 3.2 introduce this functionality?

@sabre1041 

Here is my recommendation for the implementation of this functionality:
1. The operator should introduce a new pod into the ecosystem. Example name: `quay-mirror-worker`
2. The Pod should run the Quay image but its [entrypoint](https://github.com/quay/quay/blob/master/quay-entrypoint.sh) should be `repomirror`.
3. This Pod should be ran only after the main Quay pod is finished initializing.
4. *Stretch-goal:* The ability to scale the quantity of mirror worker pods may be beneficial. 
5. Enabling this feature is typically performed in Quay's configuration application. Therefore, the operator may need to expose a CR value to enable/disable this feature.
6. For documentation, it is important to note that this feature may only be enabled by the operator during the initial installation/setup. It is possible to work-around it by modifying the `config.yaml` used by Quay but IMHO that's out of scope for the initial change.
7. If the readiness endpoint was included in Quay 3.2, it would be fantastic to use that. If not, then perhaps we mark this feature as tech-preview until we can better integrate it with Kubernetes/OpenShift.
8. I do not think there is any reason to allow Users to specify a *different* image for this pod. If different versions of Quay are ran simultaneously then unexpected results may occur.
9. Running repository mirroring under its own pod(s) should be the only supported method. We should not attempt to run repository mirroring in the primary Quay registry pod(s).
10. The Pod should mount the config and certs used by the Quay registry pod. However, it also needs the environment variable `QUAYENTRY=repomirror` set to start the correct process. This is similar in functionality to the config pod.
11. The Pod should be able to connect to postgres, redis, and the Quay registry pod(s).
